### ISSUE TITLE
Pasting large amounts of text into the terminal is very slow 

### ISSUE THREAD

**Issue by zx8** [fish-shell/fish-shell#6603]
Using https://github.com/fish-shell/fish-shell/commit/38d530bc5b18cbd98bb8ebd4858def5f89f25cf4.

---

In bash, this is instant.

In fish, this takes a painfully long time. Not only that, it's not `^C` interruptible, so I'm sometimes forced to close the entire terminal to interrupt it if I accidentally paste large amounts of text!

### Reproduction

* Run fish
* Paste the following into the terminal:

```
echo This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family. This is the documentation for fish, the friendly interactive shell. A shell is a program which helps you operate your computer by starting other programs. fish offers a command-line interface focused on usability and interactive use. Unlike other shells, fish does not follow the POSIX standard, but still roughly belongs to the same family.
```

* Observe it takes 3-5 seconds to complete the paste operation

--- 

Throughout the following clip, something even stranger happened. I was spamming `^C` to interrupt the paste, but instead of interrupting, it seems to leave me in a weird state where I was stuck mid-paste. My CPU shot up to 600% and I had to forcefully quit the terminal and start a new one to `kill` the fish process...

* https://streamable.com/j69rz

**Comment by floam** [comment]
It's probably slow highlighting. It'd be interesting to see how much better we could get this optimizing the hot path.

**Comment by zanchey** [comment]
See also #5572 (the Ctrl-C problem), #1893, #5866.

**Comment by ridiculousfish** [comment]
This is somewhat better in 3.1 since we can defer highlighting but it's still pretty bad.

**Comment by krobelus** [comment]
Will probably be fixed by #6706 although the cursor seems to be mispositioned.

For the record, `\cv` (`fish_clipboard_paste`) does not have this problem, it should always be fast.
